lib/ecto_enum.ex

```diff
@@ -57,18 +64,34 @@ defmodule EctoEnum do
     end
   end
 
+  defmodule UndeterminableStorageError do
+    defexception [:message]
+
+    def exception(kw) do
+      msg = "cannot determine storage type for #{inspect kw}"
+      %__MODULE__{message: msg}
+    end
+  end
+
   defmacro defenum(module, enum_kw) when is_list(enum_kw) do
     quote do
       kw = unquote(enum_kw) |> Macro.escape
 
+      storage = EctoEnum.storage(kw)
```

Comment:
Can't we run this code for checking the type of the data outside of the quote block? Maybe it can get run while the macro is being expanded and raise an error during compile-time.